persistence/persistence-mybatis-test/src/test/java/org/cbioportal/persistence/mybatis/GeneticProfileMyBatisRepositoryTest.java

```diff
@@ -0,0 +1,196 @@
+        Assert.assertEquals((Integer) 1, cancerStudy.getCancerStudyId());
+        Assert.assertEquals("study_tcga_pub", cancerStudy.getCancerStudyIdentifier());
+        Assert.assertEquals("brca", cancerStudy.getTypeOfCancerId());
+        Assert.assertEquals("Breast Invasive Carcinoma (TCGA, Nature 2012)", cancerStudy.getName());
+        Assert.assertEquals("BRCA (TCGA)", cancerStudy.getShortName());
+        Assert.assertEquals("<a href=\\\"http://cancergenome.nih.gov/\\\">The Cancer Genome Atlas (TCGA)</a> Breast" +
+                " Invasive Carcinoma project. 825 cases.<br><i>Nature 2012.</i> <a href=\\\"http://tcga-data.nci." +
+                "nih.gov/tcga/\\\">Raw data via the TCGA Data Portal</a>.", cancerStudy.getDescription());
+        Assert.assertEquals(true, cancerStudy.getPublicStudy());
+        Assert.assertEquals("23000897", cancerStudy.getPmid());
+        Assert.assertEquals("TCGA, Nature 2012", cancerStudy.getCitation());
+        Assert.assertEquals("SU2C-PI3K;PUBLIC;GDAC", cancerStudy.getGroups());
+        Assert.assertEquals((Integer)0 , cancerStudy.getStatus());
+    }
+
+    @Test
+    public void getAllGeneticProfilesSummaryProjection1PageSize() throws Exception {
+
+        List<GeneticProfile> result = geneticProfileMyBatisRepository.getAllGeneticProfiles("SUMMARY", 1, 0, null, null);
+
+        Assert.assertEquals(1, result.size());
+    }
+
+    @Test
+    public void getAllGeneticProfilesSummaryProjectionStableIdSort() throws Exception {
+
+        List<GeneticProfile> result = geneticProfileMyBatisRepository.getAllGeneticProfiles("SUMMARY", null, null,
+                "stableId", "ASC");
+
+        Assert.assertEquals(6, result.size());
+        Assert.assertEquals("study_tcga_pub_gistic", result.get(0).getStableId());
+        Assert.assertEquals("study_tcga_pub_log2CNA", result.get(1).getStableId());
+        Assert.assertEquals("study_tcga_pub_methylation_hm27", result.get(2).getStableId());
+        Assert.assertEquals("study_tcga_pub_mrna", result.get(3).getStableId());
+        Assert.assertEquals("study_tcga_pub_mutations", result.get(4).getStableId());
+        Assert.assertEquals("study_tcga_pub_sv", result.get(5).getStableId());
+    }
+
+    @Test
+    public void getMetaGeneticProfiles() throws Exception {
+
+        BaseMeta result = geneticProfileMyBatisRepository.getMetaGeneticProfiles();
+
+        Assert.assertEquals((Integer) 6, result.getTotalCount());
+    }
+
+    @Test
+    public void getGeneticProfileNullResult() throws Exception {
+
+        GeneticProfile result = geneticProfileMyBatisRepository.getGeneticProfile("invalid_genetic_profile");
+
+        Assert.assertNull(result);
+    }
+
+    @Test
+    public void getGeneticProfile() throws Exception {
+
+        GeneticProfile geneticProfile = geneticProfileMyBatisRepository.getGeneticProfile("study_tcga_pub_gistic");
+
+        Assert.assertEquals((Integer) 2, geneticProfile.getGeneticProfileId());
```

Comment:
why should we suppress this? Doesn't the frontend also need to know the id so that it can call related web-services that need it to find the correct genetic profile data?